eventhubs/client/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -0,0 +1,300 @@
+                properties.sharedAccessKey(), ClientConstants.TOKEN_VALIDITY);
+        } catch (InvalidKeyException | NoSuchAlgorithmException e) {
+            throw new AzureException("Could not create the EventHubSharedAccessKeyCredential.", e);
+        }
+
+        return credential(properties.endpoint().getHost(), properties.eventHubPath(), tokenCredential);
+    }
+
+    /**
+     * Sets the credential information given a connection string to the Event Hubs namespace and a path to a specific
+     * Event Hub instance.
+     *
+     * @param connectionString The connection string to use for connecting to the Event Hubs namespace; it is expected
+     *         that the shared access key properties are contained in this connection string, but not the Event Hub path.
+     * @param eventHubPath The path of the specific Event Hub to connect the client to.
+     * @return The updated EventHubClientBuilder object.
+     * @throws IllegalArgumentException if {@code connectionString} or {@code eventHubPath} is null or empty. Or, if the
+     *         {@code connectionString} contains the Event Hub path.
+     * @throws AzureException If the shared access signature token credential could not be created using the connection
+     *         string.
+     */
+    public EventHubClientBuilder credential(String connectionString, String eventHubPath) {
+        if (ImplUtils.isNullOrEmpty(eventHubPath)) {
+            throw new IllegalArgumentException("'eventHubPath' cannot be null or empty");
+        }
+
+        final ConnectionStringProperties properties = new ConnectionStringProperties(connectionString);
+        final TokenCredential tokenCredential;
+        try {
+            tokenCredential = new EventHubSharedAccessKeyCredential(properties.sharedAccessKeyName(),
+                properties.sharedAccessKey(), ClientConstants.TOKEN_VALIDITY);
+        } catch (InvalidKeyException | NoSuchAlgorithmException e) {
+            throw new AzureException("Could not create the EventHubSharedAccessKeyCredential.", e);
+        }
+
+        if (!ImplUtils.isNullOrEmpty(properties.eventHubPath())) {
+            throw new IllegalArgumentException(String.format(Locale.US,
+                "'connectionString' contains an Event Hub path [%s].  Please use the"
+                    + " credentials(String connectionString) overload. Or supply a 'connectionString' without"
+                    + " 'EntityPath' in it.", properties.eventHubPath()));
+        }
+
+        return credential(properties.endpoint().getHost(), eventHubPath, tokenCredential);
+    }
+
+    /**
+     * Sets the credential information for which Event Hub instance to connect to, and how to authorize against it.
+     *
+     * @param host The fully qualified host name for the Event Hubs namespace. This is likely to be similar to
+     *         {@literal {your-namespace}.servicebus.windows.net}.
```

Comment:
You should probably just double check that JavaDoc renders this as you expect.

Reply:
Yup! It works. Phew.